### PR TITLE
Aquisiton of token with SeCreateTokenPrivilege privilege 

### PR DESCRIPTION
**Pull request by bencikpeter**
Resolving #68 

`CageManager` is now started with a token of process that has `SeCreateTokenPrivilege` present.

To allow you to test on your systems, I´ve created a simple [PrivilegeEnumerator](https://gist.github.com/bencikpeter/d75c195c6db1048ce2175d250bc08db0) program that will aquire it parent´s process security context and enumerate all privileges available for it. Last output is a sentence saying if process has `SeCreateTokenPrivilege` or not (to spare you from searching the list).

Please try to run that code inside a cage before approving (my system is a complete mess regarding privileges after I´ve been developing that token modification library 😅 ). It will output all privileges available for `CageManager`

### CODE REVIEW

File: SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp [hunk 1]
```diff
@@ -134,24 +146,207 @@ namespace tokenLib {
...
+		auto processHandleOpt = getProcessUnderLocalSystem(privilegedProcesses);
+		if (!processHandleOpt.has_value()) {
+			std::wcout << L"Cannot locate process with needed privilege" << std::endl;
+			token = 0;
+			return false;
+		}
+
+		HANDLE processHandle = processHandleOpt.value();
+
+		HANDLE processToken = 0;
```
Comment (DonatJR) [inline SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp]: Same here: `nullptr` (there are more instances below, I'm going to stop here :D)

File: SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp [hunk 2]
```diff
@@ -134,24 +146,207 @@ namespace tokenLib {
...
+		if (!OpenProcessToken(processHandle, TOKEN_QUERY | TOKEN_DUPLICATE  | TOKEN_ASSIGN_PRIMARY, &processToken)) {
```
Comment (DonatJR) [inline SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp]: nit: unneeded additional white sapce after `TOKEN_DUPLICATE`

File: SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp [hunk 3]
```diff
@@ -134,24 +146,207 @@ namespace tokenLib {
...
+			token = 0;
+			CloseHandle(processHandle);
+			std::wcout << L"Cannot aquire token handle" << std::endl;
+			return false;
+		}
+		if (!DuplicateTokenEx(processToken, MAXIMUM_ALLOWED, NULL, SecurityImpersonation, TokenPrimary, &token))
+		{
```
Comment (DonatJR) [inline SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp]: inconsistency: you mostly open the curly bracket on the same line

File: SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp [hunk 4]
```diff
@@ -134,24 +146,207 @@ namespace tokenLib {
...
+			processes.push_back(processPid);
+		}
+		CloseHandle(processHandle);
+	}
+	if (processes.size() == 0) {
+		std::wcout << "Could not find process with SeCreateTokenPrivilege" << std::endl;
+		return std::nullopt;
+	}
+	return processes;
+
```
Comment (DonatJR) [inline SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp]: nit: unneeded empty line

File: SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp [hunk 5]
```diff
@@ -134,24 +146,207 @@ namespace tokenLib {
...
+			delete[](BYTE*) name;
+			return true;
+		}
+		delete[](BYTE*) name;
+	}
+	std::wcout << L"Selected token does not posses SeCreateTokenPrivilege" << std::endl;
+	CloseHandle(tokenHandle);
+	delete[](BYTE*) tokenPrivileges;
+	return false;
+
```
Comment (DonatJR) [inline SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp]: nit: unneeded empty line

File: SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp [hunk 6]
```diff
@@ -134,24 +146,207 @@ namespace tokenLib {
...
+			CloseHandle(processHandle);
+			CloseHandle(processToken);
+			std::wcout << L"Cannot duplicate token" << std::endl;
+			return false;
+		}
+		CloseHandle(processHandle);
+		CloseHandle(processToken);
+		return true;
+	}
+}
```
Comment (DonatJR) [inline SharkCage/SharedFunctionality/tokenLib/tokenLib.cpp]: nit: missing empty line

**Comment by bencikpeter** [comment]
@DonatJR ´s suggestions implemented, merging.

**Pull request merged**